### DISCUSSION TITLE
Gettext like text extraction or usage in combination with gettext

### DISCUSSION THREAD

**Discussion by FlorianWerndl** [s00d/nuxt-i18n-micro#15] [Q&A]
I am a huge fan of using gettext (https://jshmrtn.github.io/vue3-gettext/) within my applications as it can extract translation messages from my source code. This makes is obsolete to handle translation files by yourself and makes it easy to handle translations with tools like weblate (https://weblate.org/). As gettext is only for the messages itself it may need to be combined with tools like nuxt-i18n-mirco or i18n.nuxt . Is any gettext like feature planned in the future or do you have any advice on how it can be combined it usage? 

Thanks for your work!

**Comment by s00d** [comment]
Hello, let me know which functionality you find interesting from vue3-gettext. I might be able to create an equivalent for my module. Integrating someone else's code is very complex and time-consuming; it's easier and faster to do everything myself.

Regarding integration with other services, many services already support working with JSON. For example, I added documentation for integration with [Crowdin](https://s00d.github.io/nuxt-i18n-micro/guide/crowdin)


Regarding extracting text from code, I can basically write a simple CLI tool that can extract translations from text. However, since I couldn't find a way to get HTML during generation, it will be necessary to extract translations from the source files. There will likely be issues, and testing will be required. If you are specifically interested in this functionality, I can create the tool soon, and you can test it.

**Comment by FlorianWerndl** [comment]
The text extracting is the functionality which is interesting (it is also the feature basic feature what gettext is used for)

**Comment by s00d** [comment]
I've already developed a CLI tool that can extract text and do much more:
https://s00d.github.io/nuxt-i18n-micro/guide/cli
Some features might not work properly or may be a bit buggy since I haven't had time to thoroughly test everything. I did test the extraction feature, and it should work fine.

**Comment by FlorianWerndl** [comment]
wow. I will have a look at that when I am back from my vacation.
